### PR TITLE
2021 Dask survey results

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
This PR adds the Dask survey results and report from 2021.

The rendered notebook can be seen here: https://github.com/GenevieveBuckley/dask-examples/blob/survey2021/surveys/2021.ipynb

Cross reference: https://github.com/dask/dask-blog/pull/109

### CODE REVIEW

**Comment by GenevieveBuckley** [comment]
Couple of points:
* For some reason the separator characters are inconsistent (I'm not talking about the column separators, I'm talking about where users can select multiple options as answers for a single question. In the previous years' datasets those are separated by semicolons, but the spreadsheet I downloaded they are separated by comma-space. Some of the answer options also include commas within them, so this was a bit inconvenient). I just worked around it because that was easier, but if we need to be more strict about what data gets uploaded to this repository, please let me know.
* I included an XKCD comic ([this one](https://xkcd.com/927/)) because it felt relevant, but if that makes the tone too unprofessional/casual, let me know there too.

**Comment by GenevieveBuckley** [comment]
Also, one person complained about there being too many questions. They must have been pretty motivated to do that, since we didn't really give a good opening for that kind of feedback. They might have a point: there were 43 questions this year, the survey was roughly 50% longer than last year.

File: surveys/2021.ipynb
Comment (jacobtomlinson) [inline surveys/2021.ipynb:53]:
```suggestion
    "- To improve the resiliency of Dask (i.e. do computations complete?)\n",
```

**Review by jacobtomlinson — COMMENTED** [review]
This looks really great. Just spotted a couple of typos.

File: surveys/2021.ipynb
Comment (TomAugspurger) [inline surveys/2021.ipynb:97]:
```suggestion
      "  common = df2019.columns.intersection(df2020.columns).intersection(df2021.columns)\n"
```

File: surveys/2021.ipynb
Comment (TomAugspurger) [inline surveys/2021.ipynb:1220]: Incomplete sentence here.

**Review by TomAugspurger — APPROVED** [review]
Looks great. Thanks for working on this @GenevieveBuckley!

**Comment by GenevieveBuckley** [comment]
Thanks Tom!

File: surveys/2021.ipynb
Comment (GenevieveBuckley) [reply surveys/2021.ipynb]: Thanks for catching!
I meant to add that this was the same result as in previous surveys.

**Comment by GenevieveBuckley** [comment]
Suggestion from Matt on the blopost PR - add another graph showing that more people are saying many people widespread across their organization use Dask.

(I'll also need to rerun the notebook, to remove the FutureWarning Tom's change has just fixed)

**Comment by GenevieveBuckley** [comment]
Extra note: I saw some good quotes from people in some of the free text responses, it might be nice to add some of those to this post too.

**Comment by GenevieveBuckley** [comment]
>>  In the last year, there has been an increase in the number of people who say that many people throughout their institution use Dask (32 people said this in 2021, compared to 19 in 2020).
> Thoughts on trying to convey this visually? This seems like it might tell a nice story.

### Interesting points here:
1. From 2020 to 2021, there is a relatively substantial increase in people who say that beyond their group, many people throughout their institution use Dask. 
2. From 2019 to 2020, there was a relatively substantial drop in people who say their immediate team uses Dask.
3. The number of people using Dask mostly on their own looks pretty stable from year to year. The fluctuations appear to be explained by slightly different numbers of survey responses from year to year, not a change in the underlying data.

### Some graphs
image

This graph was generated with...
```python
q = 'Do you use Dask as part of a larger group?'
ax = sns.countplot(y=q, hue="Year", data=df.reset_index());
ax.set(ylabel="", title=q);
```

image

This graph was generated with...
```python
q = 'Do you use Dask as part of a larger group?'
ans = 'Beyond my group, many people throughout my institution use Dask'
ax = sns.countplot(x=q, hue="Year", data=df.reset_index(), order=[ans]);
```

### Why did we see these changes?
To be honest, it's not exactly clear why we saw these changes.

Re 1: the jump in people who say Dask use is widespread across an organisation

- - It doesn't seem as though there is movement from "my immediate team uses Dask" into "Dask is widespread in my organisation"
- Perhaps more people from orgs where Dask is widespread answered our survey this year? That might easily cause an over-representation if say, more people heard about it from their colleagues. But I can't test this theory because not that many people gave us information about their workplace (which is reasonable, that's pretty identifying information)
- I wondered if the existence of Coiled might be a contributing factor, but that theory does not look likely based on the data. Three people mentioned deploying with Coiled, but only one says Dask use is widespread in their org (the other two use Dask mostly on their own). I'm sure Coiled gets used for other things besides straight up deployment, but there's no mention of that in the survey results.
- For the people who did give us workplace information, the workplaces where people say Dask use is widespread mostly include universities, national labs, and Capital One. Perhaps the people at unis & research institutes are seeing more uptake of Dask by other teams?

Re 2: the drop in people who say Dask is used by their immediate team

- I don't know what could have caused this.

**Comment by GenevieveBuckley** [comment]
Ok, waiting on final approval here.

Last remaining question is, do we need this graph too (i.e. is the other one misleading without this context, or am I starting to get a bit in the weeds with details here). If yes, it would be good to talk about *why* these changes happened, but we really don't know why so `¯\_(ツ)_/¯`
graph

**Comment by GenevieveBuckley** [comment]
Update: I spoke to Matt and we're going with this graph that shows information from all years

png

Will publish it shortly.

**Comment by GenevieveBuckley** [comment]
Please merge @dask/maintenance - thank you!

I've scheduled a tweet tomorrow to promote [the blogpost](https://blog.dask.org/2021/09/15/user-survey), so it'd be really nice if the binder button works by then (it just needs merging here, then it will work).

**Comment by jacobtomlinson** [comment]
Thanks so much @GenevieveBuckley this is a great post!

**Comment by GenevieveBuckley** [comment]
Hmm, the binder badge links for all the survey notebooks are getting 404 errors - does anyone know how I can fix this?
https://github.com/dask/dask-examples/issues/196

**Comment by GenevieveBuckley** [comment]
Update: have worked out a fix for the binder links.

**Comment by jsignell** [comment]
Super interesting! Thanks for putting this together @GenevieveBuckley

**Pull request merged**
